From 8ca7e3a6e3f2ab3364319a241401f0624aa75bfc Mon Sep 17 00:00:00 2001
From: Caio Schnepper <caioschnepper@gmail.com>
Date: Wed, 25 Nov 2015 17:51:30 -0200
Subject: [PATCH 4/4] libgui: Don't assign handle to NULL after free is common

Reportedly Mali and PowerVR GPUs are crashing when setting handle to NULL
So we will set a flag for the devices that might need this aswell

Set BOARD_EGL_NEEDS_HANDLE_VALUE=true in BoardConfig.mk to use

Change-Id: I6c967f62dc6adced7583d7b2045d11cf5b25fc80
---
 libs/ui/Android.bp        | 8 ++++++++
 libs/ui/GraphicBuffer.cpp | 2 ++
 2 files changed, 10 insertions(+)

diff --git a/libs/ui/Android.bp b/libs/ui/Android.bp
index ff9d19e29..733016ae9 100644
--- a/libs/ui/Android.bp
+++ b/libs/ui/Android.bp
@@ -70,6 +70,14 @@ cc_library_shared {
     include_dirs: [
         "frameworks/native/include",
     ],
+    product_variables: {
+        lineage: {
+	    egl_needs_handle: {
+	        cppflags: ["-DEGL_NEEDS_HANDLE"],
+	    },
+        },
+    },
+
 
     shared_libs: [
         "android.hardware.graphics.allocator@2.0",
diff --git a/libs/ui/GraphicBuffer.cpp b/libs/ui/GraphicBuffer.cpp
index 254038b0a..300f1190b 100644
--- a/libs/ui/GraphicBuffer.cpp
+++ b/libs/ui/GraphicBuffer.cpp
@@ -109,7 +109,9 @@ void GraphicBuffer::free_handle()
         GraphicBufferAllocator& allocator(GraphicBufferAllocator::get());
         allocator.free(handle);
     }
+#ifndef EGL_NEEDS_HANDLE
     handle = NULL;
+#endif
 }
 
 status_t GraphicBuffer::initCheck() const {
-- 
2.11.0

